tests/DataSources/BlogPostsTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Tests\DataSources;
+
+use DateTimeImmutable;
+use Doctrine\Website\Builder\SourceFile;
+use Doctrine\Website\Builder\SourceFileParameters;
+use Doctrine\Website\Builder\SourceFileRepository;
+use Doctrine\Website\DataSource\Sorter;
+use Doctrine\Website\DataSources\BlogPosts;
+use Doctrine\Website\Tests\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use function usort;
+
+class BlogPostsTest extends TestCase
+{
+    /** @var SourceFileRepository|MockObject */
+    private $sourceFileRepository;
+
+    /** @var BlogPosts */
+    private $blogPosts;
+
+    public function testGetBlogPostsData() : void
+    {
+        $sourceFile1 = new SourceFile(
+            'md',
+            '/tmp/test1.html',
+            '/tmp/test1.html',
+            'test1',
+            new SourceFileParameters([
+                'url' => '/2018/09/02/test1.html',
+                'title' => 'Test Blog Post',
+                'authorName' => 'Jonathan H. Wage',
+                'authorEmail' => 'jonwage@gmail.com',
+            ])
+        );
+
+        $sourceFile2 = new SourceFile(
+            'md',
+            '/tmp/test.html',
+            '/tmp/test.html',
+            'test2',
+            new SourceFileParameters([
+                'url' => '/2018/09/01/test2.html',
+                'title' => 'Test Blog Post',
+                'authorName' => 'Jonathan H. Wage',
+                'authorEmail' => 'jonwage@gmail.com',
+            ])
+        );
+
+        $files = [$sourceFile1, $sourceFile2];
+
+        $this->sourceFileRepository->expects(self::once())
+            ->method('getFiles')
+            ->with('', 'source/blog')
+            ->willReturn($files);
+
+        $blogPostsData = $this->blogPosts->getData();
+
+        usort($blogPostsData, new Sorter(['date' => 'desc']));
```

Comment:
Please bring the expected result into the expected order instead of using/introducing a test-unrelated logic.

Reply:
The tested method shouldn't be sorting at all and the mock object controls in what order the result is returned, so the test shouldn't have a surprise order. A DataSource doesn't sort any values, so I think that even `array_reverse` isn't needed in `BlogPosts`. The sorting will be done by the Repository.